### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by Zainab-Alzubair**
### In this PR

1. **Add the following validations:**

_- For the User model:_
1. Name must not be blank.
2. PostsCounter must be an integer greater than or equal to zero.

_- For the Post model:_
1. Title must not be blank.
2. Title must not exceed 250 characters.
3. CommentsCounter must be an integer greater than or equal to zero.
4. LikesCounter must be an integer greater than or equal to zero.

2.  **Add unit specs for all of your models' methods and validations.**

**Thanks for reviewing my PR**

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it 'should return the recent posts' do
+    Post.create(author: subject, title: 'Hello', text: 'Hello World', commentsCounter: 0, likesCounter: 0)
+    posts = subject.recent_posts
+    expect(posts.length).to eq(1)
+  end
+
+  it 'cannot add user without name' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (codepantha) [inline spec/models/user_spec.rb:21]: - [ ] Please add one more validation test for the user that tests if posts_counter is an integer greater than or equal to zero
image

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    Post.new(author: user1, title: 'post title', text: 'post content', likesCounter: 0,
+             commentsCounter: 0)
+  end
+
+  before { subject.save }
+
+  it 'should have atitle' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (codepantha) [inline spec/models/post_spec.rb:16]: - [ ] Please test that `title` must not exceed 250 characters as per requirement 👍 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Zainab-Alzubair 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added model specs 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Zainab-Alzubair 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
